spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/HttpClientBeanPostProcessor.java

```diff
@@ -52,10 +54,10 @@ public Object postProcessAfterInitialization(Object bean, String beanName)
 				HttpTracing.class);
 		if (bean instanceof HttpClient) {
 			return ((HttpClient) bean).mapConnect(new TracingMapConnect(httpTracing))
-					.doOnRequest(TracingDoOnRequest.create(httpTracing))
-					.doOnRequestError(TracingDoOnErrorRequest.create(httpTracing))
-					.doOnResponse(TracingDoOnResponse.create(httpTracing))
-					.doOnResponseError(TracingDoOnErrorResponse.create(httpTracing));
+					.doOnRequest(new TracingDoOnRequest(httpTracing))
+					.doOnRequestError(new TracingDoOnErrorRequest(httpTracing))
+					.doOnResponse(new TracingDoOnResponse(httpTracing))
+					.doOnResponseError(new TracingDoOnErrorResponse(httpTracing));
```

Comment:
@adriancole Move `.mapConnect(new TracingMapConnect(httpTracing))` at the end as the order matters.
https://projectreactor.io/docs/core/release/reference/#_simple_context_examples

I added test in Reactor Netty to verify that behaviour
https://github.com/reactor/reactor-netty/commit/362a6b9c2c7b5514f54631bec35fa80c25c4e284

Reply:
you are awesome